File: app/src/main/java/com/finmana/app/data/SettingsStore.java
package com.finmana.app.data;

import android.content.Context;
import android.content.SharedPreferences;

import androidx.lifecycle.MutableLiveData;

public class SettingsStore {
    private static final String PREFS_NAME = "settings";
    private final SharedPreferences prefs;
    public final MutableLiveData<AppSettings> liveData = new MutableLiveData<>();

    public SettingsStore(Context context) {
        prefs = context.getApplicationContext().getSharedPreferences(PREFS_NAME, Context.MODE_PRIVATE);
        liveData.setValue(load());
    }

    public AppSettings get() {
        return load();
    }

    public void saveAi(String apiUrl, String apiKey, String model) {
        prefs.edit()
            .putString("api_url", apiUrl.trim())
            .putString("api_key", apiKey.trim())
            .putString("ai_model", model.trim())
            .apply();
        liveData.postValue(load());
    }

    public void saveGoogle(String folderId) {
        prefs.edit()
            .putString("drive_folder_id", folderId)
            .apply();
        liveData.postValue(load());
    }

    public String getSpreadsheetIdForYear(int year) {
        return prefs.getString("sheet_" + year, "");
    }

    public void saveSpreadsheetIdForYear(int year, String sheetId) {
        prefs.edit().putString("sheet_" + year, sheetId).apply();
    }

    private AppSettings load() {
        return new AppSettings(
            prefs.getString("api_url", ""),
            prefs.getString("api_key", ""),
            prefs.getString("ai_model", ""),
            prefs.getString("drive_folder_id", ""),
            ""
        );
    }
}

File: app/src/main/java/com/finmana/app/data/SettingsStore.kt
package com.finmana.app.data

import android.content.Context
import androidx.datastore.preferences.core.edit
import androidx.datastore.preferences.core.stringPreferencesKey
import androidx.datastore.preferences.preferencesDataStore
import kotlinx.coroutines.flow.Flow
import kotlinx.coroutines.flow.first
import kotlinx.coroutines.flow.map

private val Context.dataStore by preferencesDataStore("settings")

data class AppSettings(
    val apiUrl: String = "",
    val apiKey: String = "",
    val aiModel: String = "",
    val driveFolderId: String = "",
    val spreadsheetId: String = ""
)

class SettingsStore(private val context: Context) {
    private object Keys {
        val apiUrl = stringPreferencesKey("api_url")
        val apiKey = stringPreferencesKey("api_key")
        val aiModel = stringPreferencesKey("ai_model")
        val folderId = stringPreferencesKey("drive_folder_id")
        val spreadsheetId = stringPreferencesKey("spreadsheet_id")
    }

    val flow: Flow<AppSettings> = context.dataStore.data.map { p ->
        AppSettings(
            apiUrl = p[Keys.apiUrl].orEmpty(),
            apiKey = p[Keys.apiKey].orEmpty(),
            aiModel = p[Keys.aiModel].orEmpty(),
            driveFolderId = p[Keys.folderId].orEmpty(),
            spreadsheetId = p[Keys.spreadsheetId].orEmpty()
        )
    }

    suspend fun get(): AppSettings = flow.first()

    suspend fun saveAi(apiUrl: String, apiKey: String, model: String) {
        context.dataStore.edit {
            it[Keys.apiUrl] = apiUrl.trim()
            it[Keys.apiKey] = apiKey.trim()
            it[Keys.aiModel] = model.trim()
        }
    }

    suspend fun saveGoogle(folderId: String, spreadsheetId: String) {
        context.dataStore.edit {
            it[Keys.folderId] = folderId
            it[Keys.spreadsheetId] = spreadsheetId
        }
    }
}

